rotkehlchen/chain/ethereum/modules/liquity/trove.py

```diff
@@ -226,32 +240,49 @@ def _query_deposits_and_rewards(
                     usd_value=asset_price * amount,
                 ),
             )
-
         return data
 
     def get_stability_pool_balances(
             self,
             addresses: list[ChecksumEvmAddress],
-    ) -> dict[ChecksumEvmAddress, dict[str, AssetBalance]]:
-        return self._query_deposits_and_rewards(
+    ) -> dict[ChecksumEvmAddress, StabilityPoolBalance]:
+        stability_pool_balances = self._query_deposits_and_rewards(
             contract=self.stability_pool_contract,
             addresses=addresses,
             methods=('getDepositorETHGain', 'getDepositorLQTYGain', 'getCompoundedLUSDDeposit'),
             keys=('gains', 'rewards', 'deposited'),
             assets=(A_ETH, A_LQTY, A_LUSD),
         )
+        balances_with_proxy = {}
+        for addr, balances in stability_pool_balances.items():
+            balances_with_proxy[addr] = StabilityPoolBalance(
+                gains=balances['gains'],
+                rewards=balances['rewards'],
+                deposited=balances['deposited'],
+                proxy_to=self.ethereum.proxies_inquirer.proxy_to_address.get(addr),
+            )
+        return balances_with_proxy
 
     def liquity_staking_balances(
             self,
             addresses: list[ChecksumEvmAddress],
-    ) -> dict[ChecksumEvmAddress, dict[str, AssetBalance]]:
+    ) -> dict[ChecksumEvmAddress, LiquityStakingBalance]:
         """
         Query the ethereum chain to retrieve information about staked assets
         """
-        return self._query_deposits_and_rewards(
+        staking_balances = self._query_deposits_and_rewards(
             contract=self.staking_contract,
             addresses=addresses,
             methods=('stakes', 'getPendingLUSDGain', 'getPendingETHGain'),
             keys=('staked', 'lusd_rewards', 'eth_rewards'),
             assets=(A_LQTY, A_LUSD, A_ETH),
         )
+        staking_balances_with_proxy = {}
+        for addr, balances in staking_balances.items():
+            staking_balances_with_proxy[addr] = LiquityStakingBalance(
+                staked=balances['staked'],
```

Comment:
Do you need to create the class instance?  From a quick check you can avoid creating a new object and also accessing the other values and just add the `proxy_to` information. Mentioning it because this looks like unnecessary access to memory and creating objects with extra calls with minimum advantages. If you want to use the types maybe you can use

 ```
class StabilityPoolBalance(TypedDict):
    gains: AssetBalance
    rewards: AssetBalance
    deposited: AssetBalance
```

and then

```
class StabilityPoolBalanceInfo(StabilityPoolBalance):
    proxy_to: Optional[CheckSumAddress]
```

and then `_query_deposits_and_rewards` returns those two depending on the arguments and you add the field 

```
output[addr]['proxy_to'] = self.ethereum.proxies_inquirer.proxy_to_address.get(addr)
```

My concern is that you are adding extra things that complicated the logic

Reply:
Also I mention it because

```python
from typing import TypedDict


class A(TypedDict):
    name: str

def prone() -> A:
    return {
        'name': 's'
    }
```

is valid